### PR TITLE
executor: add test cases about partition table with `expression`

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve?

add test cases about partition table with `expression`

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

**Review by mjonss — COMMENTED** [review]
/LGTM
But I'm not yet a reviewer.

File: executor/partition_table_test.go [hunk 1]
```diff
@@ -489,6 +489,247 @@ func (s *partitionTableSuite) TestDynamicPruneModeWithEqualExpression(c *C) {
...
+		},
+		{
+			sql: "select * from %s where a not in (1, 5)",
+			partitions: []string{
+				"all",
+				"all",
+			},
+		},
+		{
+			sql: "select * from %s where a = 2 and a = 2",
```
[the remark this answers is not in the document]
Comment (qw4990) [reply executor/partition_table_test.go:570]: Emm, actually no, there is no such a process in our optimizer to remove duplicated conditions. 
They will be naturally handled in range calculation where all conditions will be analyzed to generate some ranges to access. And in this case, the final range is `[2, 2]`.

**Review by qw4990 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**